modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -499,3 +530,560 @@ class ProtocolBaseController extends BaseEntityController
+	template: _.template($("#EndpointListView").html())
+
+	events:
+		"click .bv_addEndpoint": "handleAddEndpointPressed"
+		"rowRemoved": "handleRowRemoved"
+		"click .bv_endpointRow": "handleEndpointRowPressed"
+		"click .bv_downloadFiles": "downloadFiles"
+	
+	rowNumberKind: "column order"
+	
+	initialize: (options) =>
+		@model = options.model
+
+		# Sort the collection by the "column order" values
+		@collection = @collection.sort (stateA, stateB) =>
+			rnA = @getRowNumberForState(stateA)
+			rnB = @getRowNumberForState(stateB)
+			return rnA - rnB
+		
+	render: => 
+		$(@el).empty()
+		$(@el).html @template()
+
+		#If the table is read-only, remove the remove and add buttons
+		if @options.readOnly == true
+			#remove column
+			@$(".bv_endpointColumnRemove").remove()
+			#remove add endpoint button
+			@$(".bv_addEndpoint").remove()
+			$(".bv_endpointManagerInstructions").hide()
+		
+		if @options.newProtocol == true
+			@$(".bv_downloadFiles").hide()
+			$(".bv_endpointManagerInstructions").hide()
+
+		#if time/units or concentration/units are disabled, remove the columns from the endpoint coontroller
+		if window.conf.protocol.endpointManager.showTime == false
+			@$(".bv_endpointColumnTime").remove()
+			@$(".bv_endpointColumnTimeUnits").remove()
+
+		if window.conf.protocol.endpointManager.showConcentration == false
+			@$(".bv_endpointColumnConcentration").remove()
+			@$(".bv_endpointColumnConcentrationUnits").remove()
+		
+		#Placeholder until we update the protocol data structure
+		# Create a list to hold the endpoint controllers in, so we can iterate through them later
+		@endpointControllers = []
+		for lsState in @collection
+			@.addOne(lsState)
+			if @options.readOnly == true
+				#hide remove buttons
+				@$(".bv_remove_row").hide()
+
+
+		if @options.readOnly == false || @options.newProtocol == true 	#Don't render associated experiments if protocol is new or readOnly
+			@getExperimentSummaryTable()
+		else #if the table isn't rendered, don't render the download files button either
+			$(".bv_downloadFiles").hide() 
+
+		@
+	
+	getExperimentSummaryTable: =>
+		#hide previously shown warnings/success text
+		$(".bv_downloadSuccess").hide()
+		$(".bv_downloadWarning").hide()
+
+		protocolCode = @model.escape('codeName')		
+		$.ajax
+			type: 'GET'
+			#there are two similar routes in ExperimentBrowserRoutes.coffee and ExperimentServiceRoutes.coffee
+			#url: "/api/experimentsForProtocol/#{protocolCode}" #ExperimentBrowserRoutes.coffee route
+			url: "/api/experiments/protocolCodename/#{protocolCode}" #ExperimentServiceRoutes.coffee route
+			success: (experiments) =>
+				@setupExperimentSummaryTable experiments
+				$(".bv_experimentTableControllerTitle").html "Experiments using " + protocolCode + ":"
+
+
+	handleEndpointRowPressed: =>
+		if @options.readOnly == false #disable logic if endpoint list controller is read only (since there won't be an experiment controller)
```

Comment:
hypernit: If you want to reduce the indentation amount, you could invert this check to something like this:
```
if @options.readyOnly
    return @
#When an endpoint row is pressed, we want to filter the experiment summary table so it only displays the experiments...
#When an endpoint row is pressed, we want to filter the experiment summary table so it only displays the experiments...
			#...with protocols that contain the endpoint, so we'll need to find the row's endpoint name and regenerate the table
#since any element within the row could be clicked on, we need to first find the parent row div and move down from there
element = $(event.target)
```

Reply:
Thats helpful, thanks!